include/nuttx/mm/vm_map.h

```diff
@@ -0,0 +1,176 @@
+/****************************************************************************
+ * include/nuttx/mm/vm_map.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __MM_VM_MAP_VM_MAP_H
+#define __MM_VM_MAP_VM_MAP_H
+
+#ifdef CONFIG_MM_VM_MAP
```

Comment:
> virtual memory maps (VM_MAP) could be used by CONFIG_MM_SHM and CONFIG_FS_RAMMAP. If taken into use, those can just "select MM_VM_MAP", and use it to store the mapping info. 

As I mention in another comment, except we add an option for mmap, otherwise CONFIG_VM_MAP always has to be enabled, which make VM_MAP redundant.

> Currently, shm has got it's own list in task group and fs_rammap has it's own global list. Those serve more or less the same purpose as what vm_map is intended for. So this can be harmonized in the future. I didn't do it in this PR yet, though.

But if without that patch, the change will make the code contain many duplicated logic, it's good chance to clean up and unify shm and mmap.

Reply:
SHM doesn't use mmap/munmap, but I added a patch for SHM, which takes the same vm_map list into use there. This in practice just changes the static mapping vector into a dynamic one.